core/src/window_service.rs

```diff
@@ -212,15 +210,15 @@ impl WindowService {
                         break;
                     }
 
-                    if let Err(e) = recv_window(
-                        bank_forks.as_ref(),
-                        leader_schedule_cache.as_ref(),
-                        &blocktree,
-                        &id,
-                        &r,
-                        &retransmit,
-                        &hash,
-                    ) {
+                    if let Err(e) = recv_window(&blocktree, &id, &r, &retransmit, &hash, |blob| {
+                        blob_filter(
+                            &id,
+                            blob,
+                            bank_forks
+                                .as_ref()
+                                .map(|bank_forks| bank_forks.read().unwrap().working_bank()),
```

Comment:
@rob-solana, I was unable to find a way to pass in a Option<&Arc<Bank>> here. Instead the arc has to be cloned. Leading to the bank.clones in the unit tests below